server/application/terminal.go

```diff
@@ -120,18 +159,21 @@ func (s *terminalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var findContainer bool
+	var foundContainerName string
 	for _, c := range pod.Spec.Containers {
 		if container == c.Name {
-			findContainer = true
+			foundContainerName = c.Name
 			break
 		}
 	}
-	if !findContainer {
-		http.Error(w, "Cannot find container", http.StatusBadRequest)
+	if foundContainerName == "" {
+		http.Error(w, "Cannot find container: "+container, http.StatusBadRequest)
```

Comment:
@crenshaw-dev - in that case we should move the log line near the top so all of those fields get logged as info no matter what

Reply:
Very good point. Having thought about it a bit more, I wonder if we should move the logging until after both RBAC checks. We could perform all the validation but save the error instead of returning it. And then _if_ the user passes the RBAC checks, we log the error as a warning, set the http error, and then return. This will make sure only authorized users with appropriate RBAC have a way to generate log lines. @leoluz do you agree, or am I being over-protective of log space?

Reply:
@crenshaw-dev I think it is good to keep in mind and protect sensitive information as much as we can. There is a different criteria however in returning data in an http response compared to internal logs. I don't think we need to be overprotective when it comes to logs (I'm not saying it's ok to log passwords :D). Even if the user is not authenticated it would still be good to log his actions from an auditing perspective.